code/game/objects/items/flashlight.dm

```diff
@@ -11,6 +11,7 @@
 	actions_types = list(/datum/action/item_action)
 	var/on = FALSE
 	var/brightness_on = 5 //luminosity when on
+	var/underbarrel_light_compatible = TRUE //Can this be turned into a underbarrel light ?
```

Comment:
Make it `FALSE` by default, and only add it to the ones that can.
Alternatively make it a flag, poke me on discord if you want to know how. If not, that's fine.

Reply:
I don't think I can, as I only want the flashlight itself to be true, and all the children are already false, and how would I set the flashlight back to true even though it's false?